client/state/data-layer/wpcom/concierge/schedules/appointments/index.js

```diff
@@ -27,6 +27,7 @@ import {
 	CONCIERGE_ERROR_NO_AVAILABLE_STAFF,
 } from 'me/concierge/constants';
 import fromApi from './from-api';
+import analytics from 'lib/analytics';
```

Comment:
This line is actually what's causing tests to break. It's been a while since I've used `lib/analytics`, but as I recall simply importing it into a module causes tests to fail unless you mock it. So I'm guessing the failing tests are importing the whole `data-layer` so this import is causing them to fail.

The solution here is actually to use [the `state/analytics` helpers](https://github.com/Automattic/wp-calypso/blob/master/client/state/analytics/README.md) — all these events are directly tied to a dispatch, so you use `withAnalytics` to attach analytics metadata to the dispatched actions, which get picked up and handled in middleware.

That should stop the CI failures, and also be better long-term.

Reply:
Good catch, I updated the code to use state helpers.